app/src/main/java/com/andrew/tasky/agenda/data/reminder/ReminderRepositoryImpl.kt

```diff
@@ -57,4 +59,48 @@ class ReminderRepositoryImpl @Inject constructor(
             )
         }
     }
+
+    override suspend fun uploadCreateAndUpdateModifiedReminders() {
+        val createRemindersDtos = db.getReminderDao().getModifiedReminders().filter {
+            it.modifiedType == ModifiedType.CREATE
+        }.map {
+            db.getReminderDao().getReminderById(it.id).toReminderDto()
+        }
+        createRemindersDtos.map { createRemindersDto ->
+            when (getAuthResult { api.createReminder(createRemindersDto) }) {
+                is AuthResult.Authorized -> {
+                    db.getReminderDao().deleteModifiedReminderById(createRemindersDto.id)
+                }
+                is AuthResult.Unauthorized -> Log.e(
+                    "SyncAgendaItem",
+                    "Unauthorized, could not create reminder"
+                )
+                is AuthResult.UnknownError -> Log.e(
+                    "SyncAgendaItem",
+                    "Unknown Error, could not create reminder"
+                )
+            }
+        }
+
+        val updateRemindersDtos = db.getReminderDao().getModifiedReminders().filter {
+            it.modifiedType == ModifiedType.UPDATE
+        }.map {
+            db.getReminderDao().getReminderById(it.id).toReminderDto()
+        }
```

Comment:
The top and bottom call are very similar and use the same data, however you read from the DB twice. You can also make use of `groupBy` which groups a list of items based on a specific key:
```
val modifiedReminders = db.getReminderDao().getModifiedReminders().groupBy { it.modifiedType }

// Get created reminders
val createdReminders = modifiedReminders[ModifiedType.CREATE]

// Get updated reminders
val updatedReminders = modifiedReminders[ModifiedType.UPDATE]
```